src/message-list/messagesActions.js

```diff
@@ -1,36 +1,33 @@
-import { Auth } from '../types';
 import { getMessages } from '../api';
 import {
+  SWITCH_NARROW,
   MESSAGE_FETCH_START,
   MESSAGE_FETCH_SUCCESS,
 } from '../constants';
 
+export const switchNarrow = (narrow) => ({
+  type: SWITCH_NARROW,
+  narrow,
+});
+
 export const fetchMessages = (
   auth,
   anchor: number,
   numBefore: number,
   numAfter: number,
   narrow,
-  isNewNarrow,
 ) =>
   async (dispatch) => {
-    dispatch({ type: MESSAGE_FETCH_START, narrow, isNewNarrow });
+    dispatch({ type: MESSAGE_FETCH_START, narrow });
 
-    const messages = await getMessages(
-      auth: Auth,
-      anchor: number,
-      numBefore: number,
-      numAfter: number,
-      narrow,
-    );
+    const messages = await getMessages(auth, anchor, numBefore, numAfter, narrow);
 
     dispatch({
       type: MESSAGE_FETCH_SUCCESS,
       auth,
       messages,
       anchor,
       narrow,
-      caughtUp: (numAfter >= numBefore &&
-        messages.length < numAfter + numBefore + (numBefore ? 1 : 0)),
+      startReached: numAfter === 0 && numBefore > messages.length,
```

Comment:
Will need to rework it in the way 'startReached' is - per cached narrow. Not much harder, but different.